examples/shopping_campaigns/get_product_bidding_category_constant.rb

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env ruby
+# Encoding: utf-8
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This example fetches the set of valid ProductBiddingCategories.
+
+require 'optparse'
+require 'google/ads/google_ads'
+
+def display_categories(categories, prefix: '')
+  categories.each do |c|
+    puts "#{prefix}#{c.fetch(:name)} [#{c.fetch(:id)}]"
+    unless c.fetch(:children).empty?
+      display_categories(
+        c.fetch(:children),
+        prefix: "#{prefix}#{c.fetch(:name)} > "
+      )
+    end
+  end
+end
+
+def get_product_bidding_category_constant(customer_id)
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  query = <<~EOD
+    SELECT
+       product_bidding_category_constant.localized_name,
+       product_bidding_category_constant.product_bidding_category_constant_parent
+    FROM
+      product_bidding_category_constant
+    WHERE
+      product_bidding_category_constant.country_code IN ('US')
+  EOD
+
+  ga_service = client.service(:GoogleAds)
+  response = ga_service.search(
+      customer_id,
+      query,
+      page_size: PAGE_SIZE,
+  )
+
+  all_categories = Hash.new do |h, k|
```

Comment:
Can you add a comment explaining that the given block provides a default if the key isn't found?

The goal of these examples is to provide simple, readable code. We need to strike a balance between doing things in a clever, succinct way and doing things in a way that even a relative beginner can quickly comprehend.